### PR TITLE
feat: add release workflow

### PR DESCRIPTION
**Pull request by getchoo**
this adds a release workflow that creates a tag as small as possible - which is great for keeping the download for the action fast. you can find an example [here](https://github.com/getchoo/attic-action/tree/v1.1.1)

it also removes the need to continuously update `dist/`, adopts standard versioning for the project, and requires 0 maintenance besides creating the release! the only drawback is that it combines index.ts and post.ts again :p

we can also see if the new nix cli fixes the issues in the test workflow...but idk /shrug

edit: *and* tests wont run on prs now since we need secrets, and prs don't have those :/

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -18,5 +18,5 @@ inputs:
 runs:
   using: "node16"
   main: "dist/index.js"
-  post: "dist/post.js"
+  post: "dist/index.js"
```
Comment (ryanccn) [inline action.yml:21]: use `post.js` >:(

File: src/index.ts [hunk 2]
```diff
@@ -1,9 +1,18 @@
 import { install } from "./stages/install";
 import { configure } from "./stages/configure";
+import { push } from "./stages/push.ts";
+import { getState, saveState } from "@actions/core";
+
+const isPost = !!getState("isPost");
```
Comment (ryanccn) [inline src/index.ts:6]: unnecessary

**Review by ryanccn — CHANGES_REQUESTED** [review]

File: action.yml (hunk 1 above)
Comment (getchoo) [reply action.yml:21]: it doesn't work with this action

File: src/index.ts (hunk 2 above)
Comment (getchoo) [reply src/index.ts:6]: can typescript really automatically convert from string to boolean? 🤔

File: action.yml (hunk 1 above)
Comment (ryanccn) [reply action.yml:21]: it does?

Comment (getchoo) [reply action.yml:21]: yeah

**Pull request merged**